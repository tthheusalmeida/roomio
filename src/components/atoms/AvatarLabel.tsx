import { mergeClassNames } from "@/utils/classNames";

interface AvatarLabelProps {
  label: string;
  data: string;
  showLabel?: boolean;
  className?: string;
}

export default function AvatarLabel({
  label,
  data,
  showLabel = true,
  className,
}: AvatarLabelProps) {
  return (
    <p className={mergeClassNames("text-violet-300", className)}>
      {showLabel && (
        <label className="font-semibold text-violet-100">{label}: </label>
      )}
      <span className="font-light text-violet-300">{data}</span>
    </p>
  );
}
