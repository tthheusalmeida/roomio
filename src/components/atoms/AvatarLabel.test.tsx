import { render, screen } from "@testing-library/react";
import AvatarLabel from "./AvatarLabel";

describe("AvatarLabel component", () => {
  it("renders with label and data by default", () => {
    const { asFragment } = render(<AvatarLabel label="Name" data="Matheus" />);

    expect(screen.getByText(/Name:/)).toBeInTheDocument();
    expect(screen.getByText("Matheus")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("does not render label when showLabel is false", () => {
    const { asFragment } = render(
      <AvatarLabel label="Name" data="Matheus" showLabel={false} />
    );

    expect(screen.queryByText(/Name:/)).not.toBeInTheDocument();
    expect(screen.getByText("Matheus")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <AvatarLabel label="Name" data="Matheus" className="extra-class" />
    );

    const p = screen.getByText(/Matheus/).closest("p");
    expect(p).toHaveClass("extra-class");
    expect(p).toHaveClass("text-violet-300");

    expect(asFragment()).toMatchSnapshot();
  });
});
