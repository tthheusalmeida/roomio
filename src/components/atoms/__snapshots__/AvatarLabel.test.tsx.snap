// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`AvatarLabel component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <p
    class="text-violet-300 extra-class"
  >
    <label
      class="font-semibold text-violet-100"
    >
      Name: 
    </label>
    <span
      class="font-light text-violet-300"
    >
      Matheus
    </span>
  </p>
</DocumentFragment>
`;

exports[`AvatarLabel component does not render label when showLabel is false 1`] = `
<DocumentFragment>
  <p
    class="text-violet-300"
  >
    <span
      class="font-light text-violet-300"
    >
      Matheus
    </span>
  </p>
</DocumentFragment>
`;

exports[`AvatarLabel component renders with label and data by default 1`] = `
<DocumentFragment>
  <p
    class="text-violet-300"
  >
    <label
      class="font-semibold text-violet-100"
    >
      Name: 
    </label>
    <span
      class="font-light text-violet-300"
    >
      Matheus
    </span>
  </p>
</DocumentFragment>
`;
